cupyx/fallback_mode/ndarray.py

```diff
@@ -0,0 +1,130 @@
+"""
+class ndarray is wrapper around cupy.ndarray
+to support fallback of methods of type `ndarray.func()`
+"""
+import numpy as np
+
+import cupy as cp
+from cupyx.fallback_mode import data_transfer
+
+try:
+    from cupyx.fallback_mode import fallback
+except ImportError:
+    import sys
+    fallback = sys.modules[__package__ + '.fallback']
+
+
+class ndarray:
+    """
+    Wrapper around cupy.ndarray
+    Gets initialized with a cupy ndarray.
+    """
+
+    def __init__(self, array):
+        self._array = array
+
+    def __getattr__(self, attr):
+        """
+        Catches attributes corresponding to ndarray.
+
+        Args:
+            attr (str): Attribute of ndarray class.
+
+        Returns:
+            (_RecursiveAttr object, self._array.attr):
+            Returns_RecursiveAttr object with numpy_object, cupy_object.
+            Returns self._array.attr if attr is not callable.
+        """
+
+        cupy_object = getattr(cp.ndarray, attr, None)
+
+        numpy_object = getattr(np.ndarray, attr)
+
+        if not callable(numpy_object):
+            return getattr(self._array, attr)
+
+        return fallback._RecursiveAttr(numpy_object, cupy_object, self)
+
+    def _get_array(self):
+        """
+        Returns _array (cupy.ndarray) of ndarray object.
+        """
+        return self._array
+
+    @property
+    def __class__(self):
+        """
+        Redirects __class__ of ndarray object to cupy.ndarray
+        Used while execution of isinstance() method.
+        """
+        return cp.ndarray
```

Comment:
In this approach, `isinstance(numpy.ndarray([1]), cupy.ndarray)` seems to be evaluated to `True`.

Reply:
Yes, because `fallback_mode.ndarray.__class__` is set up as `cupy.ndarray` for `_RecursiveAttr.__instancecheck__` to work.

Reply:
Please fix it.

Reply:
Okay, it's fixed now